### ISSUE TITLE
Ambiguity when using __declspec property (ms-extension)

### ISSUE THREAD

**Issue by dgrbic** [llvm/llvm-project#64308]
The following code snippet reproduces the issue, also on [godbolt](https://godbolt.org/z/vfoY7b73Y)
Expectation is that a property has the same rules for disambiguation as an underlying accessor method.

```
#include <map>
#include <iostream>

struct A
{
    using mapType = std::map<int,int>;
    A(mapType&& m) : valuesMap{m} {};

    __declspec (property(get = getValues)) mapType & Values;
    mapType& getValues() { return valuesMap; }
    __declspec (property(get = getValues)) mapType const& Values;
    mapType const& getValues() const { return valuesMap; }

private:
    mapType valuesMap;
};


int main()
{
    const A a{A::mapType{{1,1}, {2,2}}};

    std::cout << a.getValues().size() << "\n"; // This is OK, no ambiguity
    std::cout << a.Values.size() << "\n"; // This also should not be ambiguous, MSVC does not complain
}
```
